curriculum/challenges/english/25-front-end-development/quiz-javascript-higher-order-functions/66edcdd18a4ef8df16e6bb7e.md

```diff
@@ -17,439 +17,488 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of this code?
+
+```js
+const numbers = [1, 1, 1, 1, 1];
+let sum = 0;
+const forEachRes = numbers.forEach(num => {
+  return (sum += num);
+});
+const mapRes = numbers.map(num => {
+  return (sum += num);
+});
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+`forEachRes` is `undefined` and `mapRes` is `[1,2,3,4,5]`
 
 ---
 
-Placeholder distractor 2
+`forEachRes` is `0` and `mapRes` is `[1,2,3,4,5]`
 
 ---
 
-Placeholder distractor 3
+`forEachRes` is `5` and `mapRes` is `[1,2,3,4,5]`
 
 #### --answer--
 
-Placeholder answer
+`forEachRes` is `undefined` and `mapRes` is `[6,7,8,9,10]`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of this code?
+
+```js
+[, undefined, 'a', 'b', { 20: 5 }].sort();
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+Unsupported elements for an array to be sorted, hence error
 
 ---
 
-Placeholder distractor 2
+Callback not supplied, hence error
 
 ---
 
-Placeholder distractor 3
+```js
+[<1 empty item>, 'a', 'b', undefined, { '20': 5 }]
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+[{ '20': 5 }, 'a', 'b', undefined, empty]
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following describes a callback function in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+A function that is called immediately upon declaration.
 
 ---
 
-Placeholder distractor 2
+A function that is invoked with a specific context.
 
 ---
 
-Placeholder distractor 3
+A function that returns another function.
 
 #### --answer--
 
-Placeholder answer
+A function passed as an argument to another function, to be executed by that function's logic.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of using `reduce()` on an array?
 
 #### --distractors--
 
-Placeholder distractor 1
+A boolean indicating whether any elements meet a condition.
 
 ---
 
-Placeholder distractor 2
+An Array with all elements reduced by specified callback function
 
 ---
 
-Placeholder distractor 3
+An Array of booleans
 
 #### --answer--
 
-Placeholder answer
+It varies depending on the accumulator's initial value and the callback function.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How does the `sort()` method behave if no comparison function is provided?
 
 #### --distractors--
 
-Placeholder distractor 1
+It sorts the array as strings based on UTF-16 code units but takes undefined, empty slots and object references the beginning of the array
 
 ---
 
-Placeholder distractor 2
+It returns an error.
 
 ---
 
-Placeholder distractor 3
+It sorts the array in reverse order.
 
 #### --answer--
 
-Placeholder answer
+It sorts the array as strings based on UTF-16 code units but moves empty slots to the end of the array and object references converts to `"[object Object]"`.
```

Comment:
```suggestion
It sorts the array as strings based on UTF-16 code units. Objects are compared by converting them to the string `"[object Object]"` for the compare. Empty slots are moved to the end of the array.
```

If we can make it more clear that it is only for the string comparison that objects are converted, that might help. I have given a suggestion.